### ISSUE TITLE
`no-incorrect-template-string-interpolation` does not detect JSDoc

### ISSUE THREAD

**Issue by BridgeAR** [sindresorhus/eslint-plugin-unicorn#3424]
This is somewhat a special case where JSDoc is embedded in a string.

```js
const index =  `'use strict'

const requirePackageJson = require('${requirePackageJsonPath}')

module.exports = {
  /**
   * Load the module.
   *
   * @param {string} [id] The name/id of the module to get.
   * @returns {import('${name}') | never} The module.
   */
  get (id) { return require(id || '${name}') }
}`
```
